src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.h

```diff
@@ -61,6 +64,19 @@ class SmokeCoAlarmServer
      */
     bool SetExpressedState(chip::EndpointId endpointId, ExpressedStateEnum newExpressedState);
 
+    /**
+     * @brief Set the highest level of Expressed State according to priorityOrder
+     * @param endpointId ID of the endpoint
+     * @return true on success, false on failure
+     */
+    bool AutoSetExpressedState(chip::EndpointId endpointId);
+
+    /**
+     * @brief Set priority order
+     * @param newPriorityOrder new priority order
+     */
+    void SetPriorityOrder(std::array<ExpressedStateEnum, kPriorityOrderLength> newPriorityOrder);
```

Comment:
It's changed. Since the HandleEventTrigger also uses AutoSetExpressedState, I added emberAfHandleEventTrigger to handle  eventTrigger in the application.